### PR TITLE
Instant Patch Features

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Largely from Goon

Just some features for instant-application patches:

- Instant-application patches are now "medical patches", as such they have a different sprites
- Chem-master can now make medical patches
 - This is conditional; the patch must contain *only* pre-approved medical chems* for it to become medical

*Pre-approved chem list: antihol, charcoal, epinephrine, teporone, silver sulfadiazine, styptic, synth flesh, potassium-iodine, salbutamol, omnizine, stimulants, synaptizine, oculine, mannitol, spaceacillin, saline, salicylic acid, cryox, blood, hydrocodone, mitocholide, and rezadone.

Pic of patches:

img1

:cl: Fox McCloud
add: Chem master can produce instantaneous medical patches if all chems are safe
/:cl:

### CODE REVIEW

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 1]
```diff
@@ -426,6 +427,14 @@
 		spawn(rand(0, 15))
 			stat |= NOPOWER
 
+/obj/machinery/chem_master/emag_act(mob/user)
+	if(!safety_override)
+		to_chat(user, "<span class='notice'>You disable the safeties on the [src].</span>")
+		safety_override = 1
+	else
+		to_chat(user, "<span class='notice'>You re-enable the safeties on the [src].</span>")
```
Comment (tigercat2000) [inline code/modules/reagents/Chemistry-Machinery.dm]: isn't the "standard" for multitools to re-enable safeties and emags to disable?

Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Machinery.dm]: Not necessarily: see the defib.

**Comment by Regen1** [comment]
:-1:

Medical doesn't need more powercreep

**Comment by Kevak** [comment]
I dislike the emag function. This basically means any scitot or chemisttot can make a bunch of sarin patches and go around giving half the station sarin poisoning before anyone knows what happened.

**Comment by theColdflame** [comment]
any chems that can be negative (OD or addiction) shouldn't be whitelisted

the emag stuff's also pretty op

**Comment by Regen1** [comment]
Especially OP with infinite emags

**Comment by Fox-McCloud** [comment]
>I dislike the emag function. This basically means any scitot or chemisttot can make a bunch of sarin patches and go around giving half the station sarin poisoning before anyone knows what happened.

Oh, you mean just like para-pens, which no have message, no warning, and no indicator that it's happened at all, as opposed to a patch's very obvious "[you] forces to apply the patch to [target]".

**Comment by Kevak** [comment]
Pens have a tiny prick notice afaik. And you can't fill a pill bottle with them and run around willy nilly. You have to manually refill pens.

**Comment by tigercat2000** [comment]
@Kevak they do not have any notice

**Comment by Spacemanspark** [comment]
Pens have a tiny prick message.

**Comment by Fox-McCloud** [comment]
>Pens have a tiny prick message. 

no, they do not: https://github.com/ParadiseSS13/Paradise/blob/a3fdeaa57cdf5b84dc3be2057c0eb0373a41352a/code/modules/paperwork/pen.dm#L114

**Comment by Spacemanspark** [comment]
They used to.

**Comment by tigercat2000** [comment]
over 6 months ago

**Comment by Fox-McCloud** [comment]
They've never given a message on Paradise, ever:

https://github.com/ZomgPonies/OldParadise/blob/master/code/modules/paperwork/pen.dm#L49

Here's the olddddd olddd repository under Ponies' account, and as you can see, it wasn't there, either.

"you feel a tiny prick" is a TG thing.

**Comment by Regen1** [comment]
@Fox-McCloud 

>Oh, you mean just like para-pens, which no have message, no warning, and no indicator that it's happened at all, as opposed to a patch's very obvious "[you] forces to apply the patch to [target]".

Except a sleepy pen costs TC, and you usually only have one on you, while an instant apply patch could be stocked up in the tenfolds

**Comment by tigercat2000** [comment]
@Regen1 if you wanted non-medicinal patches you would have had to buy an emag

**Comment by Regen1** [comment]
Yes @tigercat2000, an emag that has infinite uses and tons of other application areas.

A sleepy pen has one, maybe two functions, an emag has tons of others

**Comment by IcyV** [comment]
Way too powerful in my opinion. Single click easy kills on anyone. Also completely negates the point of a hypospray.

**Comment by Krausus** [comment]
A method to instantly apply deadly reagents to someone using a tiny, seemingly-innocuous item?

This sounds suspiciously similar to syringe stabbing, which was heavily nerfed **three years ago** (39d01d4), and then, uh... accidentally removed by #3861? Huh.

**Comment by Fox-McCloud** [comment]
@Krausus 

Keep in mind, that was nerfed due to anyone having access to a syringe being able to do it; syringe stabbing was removed a number of years back, and not by @VampyrBytes 

Instant application of "deadly chems" only applies to those with traitor gear--which they can already do...with traitor gear.

**Comment by Krausus** [comment]
Oh, derp, somehow didn't notice that it was already commented out, but re-indented. It was actually removed early last year, in #352.

> Instant application of "deadly chems" only applies to those with traitor gear--which they can already do...with traitor gear.

But why do they need another, more powerful way to do it?

**Comment by theColdflame** [comment]
@Fox-McCloud an omni overdose is pretty deadly

File: code/modules/reagents/Chemistry-Machinery.dm (hunk 1 above)
Comment (monster860) [inline code/modules/reagents/Chemistry-Machinery.dm]: I thought the standard was for emags to be permanent. That way, you have a risk of detection.

**Comment by Fox-McCloud** [comment]
>omni overdose is pretty deadly

Not really. It hasn't been deadly for quite a while now (ever since addictions+OD got reworked). Back in the day though, yes, a lot of OD's were lethal as hell.

**Comment by KasparoVy** [comment]
Blob omni ODs are hilarious.

As for the sleepypen, it only gets two viable uses because it transfers 50u of horse tranquilizer each attack and only has a 100u reservoir, meaning two uses. Guaranteed to put someone in the k-hole for a good time but it's not infinite (unfortunately, it'd be dope just running about knocking everyone out)

Can't wait to get instakilled by a memer with an emag and a nice deathmix

**Comment by Bigtimetripod** [comment]
Ye people here complaining that this will replace Para pen... no it will not, a d para pen has massive amount of uses. 

Healing para-pen: Have a beaker full of omnizine with ye, no chemmaster? Transfer 1-3 units for minor healing with the help of a pipette, inject with parapen, but faster. If ye know yer gonna be in combat, inject 30 units instantly for super robust healing. Take it up a level! Cryoxodone, oxygen and cryo(cold thingy that reacts with oxygen

**Comment by Bigtimetripod** [comment]
And my phone sent half the text only...

**Comment by Bigtimetripod** [comment]
Well.. death with parapen can be achieved easy, since 10 units of sarin is enough to kill anyone, have a beaker with 100 units of sarin, and fill pen up to only 10 units. This way parapen can stealth inject and KILL 10 people with one large beaker, bluespace beaker? 30 people... without breakin a sweat!... Be warned, injected people will see they are twitching and know ye did something, so try to outrun em for a minute.

 I mean, i got pissed for hearing parapen has one or two uses... it can be the most robust melee weapon in the game, IF ye know how to use it! Its a one hit kill, stun, burn, sleep etc. weapon if ye out effort into chems ye inject people with.

**Comment by Dumbdumn5** [comment]
Deathsprays are dangerous enough for the crafty science/chemistry traitor, and that's already near instant and nigh unavoidable if the player has good aim. Instantly applied patches, while obvious, would possibly be even worse.

**Comment by Fox-McCloud** [comment]
Removed meth from the safe list and removed emagging of the machine/patches.

Instant medical patches are still a thing.

**Comment by IcyV** [comment]
I'll personally agree with it if the approved list is tapered down some more. Still seems stupidly powerful. Someone trying to gun me down with a laser? Well, good thing I have like 30 synth-flesh patches with me.

Obviously depends on the amount of reagent allowed in these, of course.

**Comment by Krausus** [comment]
This seems fine now, though I won't be surprised when someone complains about one person healing another person too quickly.

> Someone trying to gun me down with a laser? Well, good thing I have like 30 synth-flesh patches with me.

You already instantly use all pills and patches on yourself, so unless you're trying to synth-flesh the guy shooting you...

**Comment by IcyV** [comment]
Yeah, but with this it also applies to other people as well. Guy next to me is getting shot? 30 patches ahoy. Insta-heals everywhere on everyone. It puts mediguns and hyposprays to shame.

**Comment by Krausus** [comment]
Well, yes, if you go through the trouble to make 30 synthflesh patches, you can do a lot of healing. Even if you have to spend the current 3 seconds applying each one, you'll be healing up to 20 brute/burn per second. But if you're going through all this trouble anyway, you could just prepare instant synthflesh smoke mixes. Or instant synthflesh smoke grenades. A lot of things in chemistry are ridiculous if you spend enough time doing the prep work. And it just got merged while I was typing this, so here we goooo~

**Comment by IcyV** [comment]
The quick merges with major disagreement too stronk.

**Comment by Spacemanspark** [comment]
I think they discuss it internally before merging it.

**Comment by TheDZD** [comment]
Most of the complaints were about the emag system, some of the concerns are invalid (ODs aren't anywhere near as horrible as they used to be, and you can't get addicted via patches). All of two people complained about these, one of whom is quite literally now grasping at straws over mechanics he doesn't properly understand.

**Comment by IcyV** [comment]
>one of whom is quite literally now grasping at straws over mechanics he doesn't properly understand.

Professional and classy. Totally isn't that I spaced out and left out a single sentence of my initial argument. Totally because I have 0 understanding of a game I've been playing for many years and knew extremely well at a point in time.

**Comment by Bigtimetripod** [comment]
I was soo hyped for the emagged option... guess ill just stick to my parapen...

**Pull request merged**